ccdproc/image_collection.py

```diff
@@ -330,7 +351,22 @@ def _get_files(self):
             else:
                 files = self._filenames
         else:
-            files = self._fits_files_in_directory()
+            _files = self._fits_files_in_directory()
+
+            files = []
+            for fn in _files:
```

Comment:
Personally I prefer list comprehensions because the logic is more straight-forward:

```
if self.glob_include is not None:
    files = fnmatch.filter(files, self.glob_include)
if self.glob_exclude is not None:
    files = [file for file in files if not fnmatch.fnmatch(fn, self.glob_exclude)]
```

It's probably also faster because it doesn't need to check the `self.glob_* is not None` each iteration and [`fnmatch.filter`](https://docs.python.org/library/fnmatch.html#fnmatch.filter) advertises itself as "being faster".